src/use-case/my-script.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+# BASH3 Boilerplate
+#
+# This file:
+#
+#  - Is a template to write better bash scripts
+#  - Is delete-key friendly, in case you don't need e.g. command line option parsing
+#
+# More info:
+#
+#  - https://github.com/kvz/bash3boilerplate
+#  - http://kvz.io/blog/2013/02/26/introducing-bash3boilerplate/
+#
+# Version: 2.0.0
+#
+# Authors:
+#
+#  - Kevin van Zonneveld (http://kvz.io)
+#  - Izaak Beekman (https://izaakbeekman.com/)
+#  - Alexander Rathai (Alexander.Rathai@gmail.com)
+#  - Dr. Damian Rouson (http://www.sourceryinstitute.org/) (documentation)
+#
+# Usage:
+#
+#  LOG_LEVEL=7 ./main.sh -f /tmp/x -d
+#
+# Licensed under MIT
+# Copyright (c) 2013 Kevin van Zonneveld (http://kvz.io)
+
+# Incorporate the bash3boilerplate content that does not need to change from one use to the next
+# and designate the name of the file containing the current file's usage as the file with "-usage"
+# appended to the current file's name:
+# name followed by "-usage"
+__filename="$(basename "${BASH_SOURCE[0]}")"
+. bootstrap.sh -f ${__filename}-usage
+
+# Command-Line Interface (CLI) 
+# ----------------------------
+# This 'while' block below reads the file named in usage_page, which will also be used to 
+# parse CLI options, their associated arguments, their default values, and their status as
+# required or optional.  See the above-referenced bootstrap script for a comprehensive 
+# example of usage page contents.  See my-script.sh-usage for the specific usage page for
+# the current script.
+while IFS='' read -r line || [[ -n "$line" ]]; do
+  usage="${usage:-}""${line}"$'\n'
```

Comment:
I think this is a pretty compelling story/use case. I am short on time, but think that if @kvz doesn't want to take this module/libary approach, it might be worthwhile to fork the project to enable that....  I think that there is some simplifying and cleaning that could be done to  reduce the complexity of the refactored code, but agree that the more static portions would be great to be able to drop in and reuse without modification.